packages/tabs/src/TabTitle/TabTitle.tsx

```diff
@@ -15,36 +30,60 @@ import {
 import { BaseTabTitleProps } from './TabTitle.types';
 
 const TabTitle: ExtendableBox<BaseTabTitleProps, 'button'> = ({
-  selected = false,
-  disabled = false,
   children,
   className,
   darkMode,
-  parentRef,
+  disabled = false,
+  onClick,
+  selectedIndex,
   ...rest
 }: BaseTabTitleProps) => {
-  const titleRef = useRef<HTMLAnchorElement | HTMLButtonElement>(null);
   const baseFontSize: BaseFontSize = useUpdatedBaseFontSize();
+  const titleRef = useRef<HTMLAnchorElement | HTMLButtonElement>(null);
+  const { index, ref, id } = useDescendant(TabsDescendantsContext);
+  const { descendants: tabDescendants } = useDescendantsContext(
+    TabsDescendantsContext,
+  );
+  const { descendants: tabPanelDescendants } = useDescendantsContext(
+    TabPanelsDescendantsContext,
+  );
 
   const theme = darkMode ? Theme.Dark : Theme.Light;
+  const selected = index === selectedIndex;
 
   // Checks to see if the current activeElement is a part of the same tab set
   // as the current TabTitle. If so, and the current TabTitle is not disabled
   // and is selected, we manually move focus to that TabTitle.
   useEffect(() => {
-    const tabsList = Array.from(parentRef?.children ?? []);
+    const tabList = tabDescendants.map(
+      descendant => descendant.element.parentElement,
+    );
```

Comment:
I feel like there's a better way to do this logic now

Reply:
We're essentially responding to changes to the `selected` prop here, so it would be better to run these side effects when we change that state as opposed to firing a complex effect like this.

One quick solution would be in `Tabs.tsx` to write:
```jsx
  const updateSelection = useCallback(
    (index: number) => {
      setSelected?.(index);
      tabDescendants[index].element.parentElement?.focus();
    },
    [setSelected, tabDescendants],
  );
]
```
and then call `updateSelection` in place of `setSelected`. 

One thing this doesn't account for is when `selected` is controlled, so we may need a custom hook to account for both these possibilities (e.g. a smaller `useEffect` in `Tabs.tsx` that responds to changes in the `controlledSelected` prop state)

Reply:
We can't access the `tabDescendants` in `Tabs.tsx` because it's outside of the provider. I think it fits appropriately where it is, but I rewrote it to make it more easily readable

Reply:
ok I think this is looking better now. let me know what you think!